wn/lmf.py

```diff
@@ -0,0 +1,639 @@
+    (f'{{{_dc_uri}}}identifier', 'identifier'),
+    (f'{{{_dc_uri}}}publisher', 'publisher'),
+    (f'{{{_dc_uri}}}relation', 'relation'),
+    (f'{{{_dc_uri}}}rights', 'rights'),
+    (f'{{{_dc_uri}}}source', 'source'),
+    (f'{{{_dc_uri}}}subject', 'subject'),
+    (f'{{{_dc_uri}}}title', 'title'),
+    (f'{{{_dc_uri}}}type', 'type'),
+)
+
+DublinCoreMetadata = Dict  # While we support Python < 3.8
+
+# use the following if Python 3.8+
+# class DublinCoreMetadata(TypedDict, total=False):
+#     contributor: str
+#     coverage: str
+#     creator: str
+#     date: str
+#     description: str
+#     format: str
+#     identifier: str
+#     publisher: str
+#     relation: str
+#     rights: str
+#     source: str
+#     subject: str
+#     title: str
+#     -type: str  # NOTE: -type is to avoid mypy type comments
+
+
+# These types model the WN-LMF DTD
+# http://globalwordnet.github.io/schemas/WN-LMF-1.0.dtd
+
+class Count(NamedTuple):
+    value: int
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class SyntacticBehaviour(NamedTuple):
+    subcategorization_frame: str
+    senses: Tuple[str, ...]
+
+
+class SenseRelation(NamedTuple):
+    target: str
+    type: str  # Literal[*_sense_rels] if python 3.8+
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class SynsetRelation(NamedTuple):
+    target: str
+    type: str  # Literal[*_synset_rels] if python 3.8+
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class Example(NamedTuple):
+    text: str
+    language: Optional[str] = None
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class ILIDefinition(NamedTuple):
+    text: str
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class Definition(NamedTuple):
+    text: str
+    language: Optional[str] = None
+    source_sense: Optional[str] = None
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class Synset(NamedTuple):
+    id: str
+    ili: str
+    pos: Optional[str] = None  # Literal[*_pos] if Python 3.8+
+    definitions: Tuple[Definition, ...] = ()
+    ili_definition: Optional[ILIDefinition] = None
+    relations: Tuple[SynsetRelation, ...] = ()
+    examples: Tuple[Example, ...] = ()
+    status: Optional[str] = None
+    note: Optional[str] = None
+    confidence: Optional[float] = None
+    dcmeta: Optional[DublinCoreMetadata] = None
+
+
+class Sense(NamedTuple):
```

Comment:
Looping definitions? Synsets -> Sense -> Synset -> Sense ?? 

Hmmm okay, I've to look at LMF more carefully.

Reply:
Where's the loop? In LMF, LexicalEntries (i.e., Lemmas and asscoiated info) list their Senses which link them to Synsets via the Synsets' ids.